salt/modules/system.py

```diff
@@ -608,3 +609,54 @@ def get_computer_name():
         salt '*' network.get_hostname
     '''
     return __salt__['network.get_hostname']()
+
+
+NILRT_REBOOT_WITNESS_PATH = '/var/volatile/tmp/salt/reboot_witnessed'
+
+NILRT_FAMILY_VERIFY_CMD = 'sh -c \'lsb_release -d | grep -qi "NI Linux Real-Time"\''
+
+
+@depends(NILRT_FAMILY_VERIFY_CMD, retcode=0)
+def set_reboot_required_witnessed():
+    '''
+    This function is used to remember that an event indicating that a reboot is
+    required was witnessed. This function writes to a temporary filesystem so
+    the event gets cleared upon reboot.
+
+    Returns:
+        bool: ``True`` if successful, otherwise ``False``
+
+    .. code-block:: bash
+
+        salt '*' system.set_reboot_required_witnessed
+    '''
+    dir_path = os.path.dirname(NILRT_REBOOT_WITNESS_PATH)
+    if not os.path.exists(dir_path):
+        try:
+            os.makedirs(dir_path)
+        except OSError as ex:
+            raise SaltInvocationError('Error creating {0} (-{1}): {2}'
+                                      .format(dir_path, ex.errno, ex.strerror))
+
+        __salt__['cmd.run']('touch {0}'.format(NILRT_REBOOT_WITNESS_PATH))
+        return True
```

Comment:
And if that fails for whatever reason with error code non zero, you're still saying "all perfect". I would not return it here, but at the end, explicitly that the exit code is 0 and then bool it:

```python
errcode = -1
...
if not os.path.exists(dir_path):
    ...
    errcode = ....

return errcode == 0
```

Reply:
Done.